cmd/metrics_exporter/metrics_exporter.go

```diff
@@ -0,0 +1,356 @@
+			logrus.Fatalf(ybmAuthClient.GetApiErrorDetails(err))
+		}
+
+		fmt.Printf("Removing associated Metrics Exporter Config from cluster %s\n", formatter.Colorize(clusterName, formatter.GREEN_COLOR))
+	},
+}
+
+var associateMetricsExporterWithClusterCmd = &cobra.Command{
+	Use:   "attach",
+	Short: "Associate Metrics Exporter Config with Cluster",
+	Long:  "Associate Metrics Exporter Config with Cluster",
+	Run: func(cmd *cobra.Command, args []string) {
+		authApi, err := ybmAuthClient.NewAuthApiClient()
+		if err != nil {
+			logrus.Fatalf("could not initiate api client: %s", err.Error())
+		}
+		authApi.GetInfo("", "")
+
+		clusterName, _ := cmd.Flags().GetString("cluster-name")
+		clusterId, err := authApi.GetClusterIdByName(clusterName)
+		if err != nil {
+			logrus.Fatal(err)
+		}
+
+		configName, _ := cmd.Flags().GetString("config-name")
+
+		configId, err := authApi.GetConfigIdByName(configName)
+		if err != nil {
+			logrus.Fatalf(ybmAuthClient.GetApiErrorDetails(err))
+		}
+
+		metricsExporterClusterConfigSpec := ybmclient.NewMetricsExporterClusterConfigurationSpec(configId)
+
+		_, r, err := authApi.AssociateMetricsExporterWithCluster(clusterId).MetricsExporterClusterConfigurationSpec(*metricsExporterClusterConfigSpec).Execute()
+
+		if err != nil {
+			logrus.Debugf("Full HTTP response: %v", r)
+			logrus.Fatalf(ybmAuthClient.GetApiErrorDetails(err))
+		}
+
+		fmt.Printf("Attaching Metrics Exporter Config %s with cluster %s\n", formatter.Colorize(configName, formatter.GREEN_COLOR), formatter.Colorize(clusterName, formatter.GREEN_COLOR))
+	},
+}
+
+var stopMetricsExporterCmd = &cobra.Command{
+	Use:   "pause",
+	Short: "Stop Metrics Exporter",
+	Long:  "Stop Metrics Exporter",
+	Run: func(cmd *cobra.Command, args []string) {
+		authApi, err := ybmAuthClient.NewAuthApiClient()
+		if err != nil {
+			logrus.Fatalf("could not initiate api client: %s", err.Error())
+		}
+		authApi.GetInfo("", "")
+
+		clusterName, _ := cmd.Flags().GetString("cluster-name")
+		clusterId, err := authApi.GetClusterIdByName(clusterName)
+		if err != nil {
+			logrus.Fatal(err)
+		}
+
+		r, err := authApi.StopMetricsExporter(clusterId).Execute()
+
+		if err != nil {
+			logrus.Debugf("Full HTTP response: %v", r)
+			logrus.Fatalf(ybmAuthClient.GetApiErrorDetails(err))
+		}
+
+		fmt.Printf("Stopping Metrics Exporter for cluster %s\n", formatter.Colorize(clusterName, formatter.GREEN_COLOR))
+	},
+}
+
+var updateMetricsExporterCmd = &cobra.Command{
+	Use:   "update",
+	Short: "Update Metrics Exporter Config",
+	Long:  "Update Metrics Exporter Config",
+	Run: func(cmd *cobra.Command, args []string) {
+		authApi, err := ybmAuthClient.NewAuthApiClient()
+		if err != nil {
+			logrus.Fatalf("could not initiate api client: %s", err.Error())
+		}
+		authApi.GetInfo("", "")
+
+		metricsExporterName, _ := cmd.Flags().GetString("config-name")
+		metricsSinkType, _ := cmd.Flags().GetString("type")
+		datadogSpecString, _ := cmd.Flags().GetStringToString("datadog-spec")
```

Comment:
Validate that the flag was set before trying to use it.

Reply:
Refactor this parts as well @cdavid